core/constant_inc.php

```diff
@@ -623,4 +623,12 @@
 # Note: mantis ids are defined in schema as "I UNSIGNED", which Adodb maps to
 # the closest integer (4 bytes) type available. As some DBs dont support unsigned
 # types, 2^31 is a safe limit to be used for all.
-define( 'DB_MAX_INT', 2147483647 );
\ No newline at end of file
+define( 'DB_MAX_INT', 2147483647 );
+
+# Databse functional type identifiers.
+define( 'DB_TYPE_UNDEFINED', 0 );
+define( 'DB_TYPE_MYSQL', 1 );
```

Comment:
Should we have all database api except db_get_type() use these constants instead of comparing directly against the config option?  For example, db_query seems to still act directly on the config option.